### DISCUSSION TITLE
How hard would it be to export the style and file export functionality

### DISCUSSION THREAD

**Discussion by FonixAU** [naturalcrit/homebrewery#3864] [Q&A]
I made a simple site for my house-hold to export a spell book dynamically through the dnd5e api, but its very basic and honestly pretty ugly, I just made it in my lunch break to help my group out since a lot of the players were new and or also new to their classes so they had to keep passing the PHB around and checking stuff. This is meant as a replacement to spell cards.

Now I wanted to pretty-fy it, and seeing how you guys construct the pages so cleanly I really wanted to use some of that logic since in essence I am just making html blocks out of json data, but since I am still a novice when it comes to development, I feel really out of my depth and intimidated when trying to understand what is going on in the background on your side when it comes to styling.

How would I go about trying to use some of the PBH fonts, colours and frames, new-page identification and so on for my spell book generator?

**Comment by 5e-Cleric** [comment]
Okay, let's see, If you are generating HTML through an api, you could take bits and parts of our CSS and use it to style them, definitely.

It will all come down to knowing how CSS works, so i would recommend you take a look at, at least 1 basic tutorial, it is pretty straightforward.

All our assets are in this repository, and you can access them, i recommend using a tool to download entire directories like [Download Directory](https://download-directory.github.io/), and inputing our assets folder (https://github.com/naturalcrit/homebrewery/tree/master/themes/assets), then maybe also the 5e fonts folder (https://github.com/naturalcrit/homebrewery/tree/master/themes/fonts/5e).

  With that and checking out our CSS, you could start, or, depending on your HTML output, you may have an easier time starting from scratch, using the inspector tool in our website, to see where we apply everything.

I think our api endpoints are open, so you could also probably use those sending markdown to generate html, but i am not sure how easy that will be.

**Reply by FonixAU** [reply]
By Ooo the endpoint part sounds interesting. I tried what you described basically as soon as I posted the message, the issue I’m getting is that the masking isn’t working (the grayscale image comes up but the colouring doesn’t happen) and the border image isn’t loading into the pdf. Even though it renders fine when I just print it to the site. The styling I can probably get working well enough but exporting everything so cleanly like it does on your website is probably going to be my biggest challenge. I’m currently trialing making it into react components and trying to use a library to export them instead but not sure how well that will work out. I am beggining to wonder if it would have been faster for me to make all of the spell books manually 💀🤣
maybe once I’m at my desk I’ll make the repo public or share some images

**Reply by 5e-Cleric** [reply]
I can take a look, but i am no printing genius, if it is a CSS issue i should be able to help
